### PR TITLE
Single page app

### PR DESCRIPTION
**Pull request by andrewhuang**
Pulls in [ui-router](https://ui-router.github.io/) which handles front end routing (mapping urls to components and handles data binding). This allows us to avoid using Django to serve the details page and also sets us up for easily adding new pages in the future solely on the frontend (i.e. settings page, advanced analysis, etc.)

### CODE REVIEW

File: cal/cal/templates/category_detail.html [hunk 1]
```diff
@@ -1,9 +0,0 @@
-{% extends "base.html" %}
-
-{% block content %}
-<div class="container">
-  <h1>Category: "{{ category.label }}"</h1>
-  <p>Cumulative hours: {{ category.hours }}</p>
```
Comment (jamesuejio) [inline cal/cal/templates/category_detail.html:6]: Add the category name and cumulative hours back to the details page. I know it's a minor detail but just want to make sure this is easily doable.

File: cal/cal/templates/tag_detail.html [hunk 2]
```diff
@@ -1,9 +0,0 @@
-{% extends "base.html" %}
-
-{% block content %}
-<div class="container">
-  <h1>Tag: "{{ tag.label }}"</h1>
-  <p style="margin-left: 15px">Keyword(s): {{ tag.keywords }}<p>
```
Comment (jamesuejio) [inline cal/cal/templates/tag_detail.html:6]: Add Tag and keywords back to details page.

**Review by jamesuejio — CHANGES_REQUESTED** [review]
Everyone looks good except to include more details into details pages.

**Comment by andrewhuang** [comment]
@jalagar: Done. PTAL

File: cal/cal/static/js/components/category-details.js [hunk 3]
```diff
@@ -11,14 +11,14 @@ analyticsApp.component('categoryDetails', {
 
 function CategoriesDetailCtrl($scope, $http, QueryService){
   var _this = this;
-  this.$onInit = () => {
+  this.$onInit = function () {
```
Comment (jamesuejio) [inline cal/cal/static/js/components/category-details.js:14]: ty

**Review by jamesuejio — APPROVED** [review]

**Pull request merged**